src/main/java/org/broadinstitute/hellbender/engine/spark/SparkSharder.java

```diff
@@ -296,11 +299,14 @@ public T apply(@Nullable Tuple2<I, Iterable<L>> input) {
                 addPartitionReadExtent(extents, partitionIndex, current.getContig(), current.getStart(), next.getStart() + maxLocatableLength);
             } else {
                 // complete current contig
-                int contigEnd = sequenceDictionary.getSequence(current.getContig()).getSequenceLength();
+                SAMSequenceRecord seq = sequenceDictionary.getSequence(current.getContig());
+                Utils.nonNull(seq, "Contig not found in sequence dictionary: " + current.getContig());
```

Comment:
`Utils.nonNull()` will throw an `IllegalArgumentException`, which is not ideal. Could you replace calls to `Utils.nonNull()` with calls to `Utils.validate()`, which throws a more appropriate `IllegalStateException`?